### PR TITLE
Build a Markdown XBlock based on the HTML XBlock by OpenCraft

### PR DESCRIPTION
**Pull request by mrtmm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-jQuery being present, and it allows users to include arbitrary JavaScript).
+This XBlock allows course authors to create and edit course content in Markdown
+and displays it as HTML.
 
 ## Installation
-You may install XBlock-html using its setup.py, or if you prefer to use pip, run:
+You may install the markdown-xblock using its setup.py, or if you prefer to use pip, run:
 
 ```shell
 pip install https://github.com/open-craft/xblock-html
```
Comment (fghaas) [inline README.md:12]: You probably want to fix this. :)

File: markdown_xblock/html.py [hunk 2]
```diff
@@ -2,53 +2,78 @@
...
+DEFAULT_EXTRAS = [
+    "code-friendly",
+    "fenced-code-blocks",
+    "footnotes",
+    "tables",
+    "use-file-vars"
+]
+DEFAULT_SETTINGS = {
+    "extras": DEFAULT_EXTRAS
+}
```
Comment (fghaas) [inline markdown_xblock/html.py:30]: Missing `safe_mode: True` here?

File: markdown_xblock/html.py [hunk 3]
```diff
@@ -2,53 +2,78 @@
...
+        default=False,
+        scope=Scope.content
     )
-    editable_fields = ('display_name', 'editor', 'allow_javascript')
+    editor = 'markdown'
+
+    @property
+    def editable_fields(self):
+        """The property for editable fields."""
+        if self.runtime._user.is_staff:
```
Comment (fghaas) [inline markdown_xblock/html.py:73]: I'm not sure I like this bit. This appears to allow a `staff` user to decide whether inline HTML is allowed or not. Staff users and course authors are often identical groups. That means that I as a site operator can't decide whether I trust course authors to not inject malicious or broken code; it's course authors that decide for themselves. Which makes the toggle kind of useless.

So I'd rather see this set via `settings` (which the site operator, not a course author, controls). Or have a toggle in `settings` which defines whether staff users can set this.

**Review by fghaas — CHANGES_REQUESTED** [review]
This looks great in general, I do have some thoughts about how you decided to toggle `safe_mode` though. :)

File: markdown_xblock/html.py (hunk 3 above)
Comment (mrtmm) [reply markdown_xblock/html.py:73]: Ok, that makes sense, I'll do that.

File: setup.py [hunk 4]
```diff
@@ -22,22 +22,22 @@ def package_data(pkg, roots):
 
 
 setup(
-    name='html-xblock',
+    name='markdown-xblock',
     version='1.0.0',
     description='HTML XBlock will help creating and using a secure and easy-to-use HTML blocks',
     license='AGPL v3',
```
[the remark this answers is not in the document]
Comment (fghaas) [reply setup.py:28]: Again for PyPI purposes, add:
```python
    classifiers=[
        'Development Status :: 4 - Beta',
        'Framework :: Django',
        'Intended Audience :: Education',
        'License :: OSI Approved :: GNU Affero General Public License v3',
        'Topic :: Education :: Computer Aided Instruction (CAI)',
        'Topic :: Education',
    ],
```

**Review by fghaas — COMMENTED** [review]
Looks good! I just have a few minor suggestions, mostly around packaging.

**Comment by mrtmm** [comment]
Thanks @fghaas, I made the changes.

**Comment by fghaas** [comment]
@mrtmm One more suggestion for you, in 71546f0 — let me know what you think.

**Comment by mrtmm** [comment]
Looks good to me.

**Pull request merged**